public/EditorPage.tsx

```diff
@@ -0,0 +1,848 @@
+                    >
+                        <Tooltip
+                            content={"Copy JSON configuration to clipboard"}
+                            placement="bottom"
+                        >
+                            <ClipboardText
+                                size={24}
+                            />
+                        </Tooltip>
+                        <ToastContainer position="bottom-right" />
+                    </span>
+                    <span
+                        className="hover:bg-slate-50 hover:text-slate-900 cursor-pointer rounded-full p-1 m-0.5"
+                        onClick={onOpen}
+                    >
+                        <Tooltip
+                            content={"Paste JSON to draw face"}
+                            placement="bottom"
+                        >
+                            <UploadSimple
+                                size={24}
+                            />
+                        </Tooltip>
+                    </span>
+                    <span
+                        className="hover:bg-slate-50 hover:text-slate-900 cursor-pointer rounded-full p-1 m-0.5"
+                        onClick={async () => { await DownloadSvgAsPng(faceConfig) }}
+                    >
+                        <Tooltip
+                            content={"Download face as PNG image"}
+                            placement="bottom"
+                        >
+                            <DownloadSimple
+                                size={24}
+                            />
+                        </Tooltip>
+                    </span>
+                </div>
+            </div>
+            <div className="  font-bold w-screen flex  items-start ">
+                <EditorPageToolbar />
+                <div className="flex flex-col-reverse md:flex-row items-center justify-around w-5/6">
+                    <EditorPageGallery />
+                    <MainFaceDisplay />
+                </div>
+            </div>
+            <Modal
+                className="w-1/2"
+                shadow="md"
+                size="xl"
+                isOpen={isOpen}
+                onOpenChange={onOpenChange}
+            >
+                <ModalContent>
+                    {(_) => (
+                        <>
+                            <ModalHeader className="flex flex-col gap-1">Paste JSON to Render Face</ModalHeader>
+                            <ModalBody>
+                                <Textarea
+                                    value={textAreaValue}
+                                    isInvalid={!textAreaValid}
+                                    ref={textRef}
+                                    // errorMessage={!textAreaValid ? ("Invalid JSON. Refer to the <a src='https://www.json.org/json-en.html'>JSON spec</a>") : null}
+                                    errorMessage={!textAreaValid ? errorMessage : null}
```

Comment:
For error handling here, get rid of the "Validate" button. Just make validation run when you click "Draw" and keep the window open if it's not valid. I don't think anybody will care about validation errors until they're ready to draw.